framework/src/base/FEProblem.C

```diff
@@ -3129,6 +3130,16 @@ void FEProblem::restoreSolutions()
     _displaced_problem->updateMesh(*_nl.currentSolution(), *_aux.currentSolution());
 }
 
+void
+FEProblem::outputStep(ExecFlagType type)
```

Comment:
This makes me very sad, I tried very hard to keep FEProblem out of the output mix. I would really like all output related stuff go through the warehouse. If we do this, we need to move other functions as well to keep the API consistent. I don't want to see calls to both the OutputWarehouse and FEProblem from things, it should be one or the other.